src/rpc/mod.rs

```diff
@@ -112,3 +125,29 @@ async fn make_rpc_call<T: Serialize, R: for<'de> Deserialize<'de>>(
         }
     }
 }
+
+async fn make_retrying_rpc_call<T: Serialize, R: for<'de> Deserialize<'de>>(
+    params: &T,
+    timeout: Option<u64>,
+    max_retries: u32,
+) -> Result<R> {
+    let mut attempts = 0;
+    loop {
+        match make_rpc_call(params, timeout).await {
+            Ok(result) => return Ok(result),
+            Err(e) => {
+                attempts += 1;
+                if attempts > max_retries {
+                    warn!("Operation failed with error: {:?}. Max retries reached", e);
+                    return Err(e);
+                }
+                let backoff = Duration::from_secs(2_u64.pow(attempts));
```

Comment:
better if we avoid potential unbounded backoff duration by setting a max duration eg:

`let backoff = Duration::from_secs(2_u64.pow(attempts.min(10)));`

Reply:
Ready to merge once the above is addressed